pkg/kv/iterators_test.go

```diff
@@ -27,3 +28,16 @@ func TestPartitionIterator_ClosedBehaviour(t *testing.T) {
 		t.Fatal("Err() on closed iterator", err)
 	}
 }
+
+func TestPartitionIterator_CloseAfterSeekGEFailed(t *testing.T) {
+	ctx := context.Background()
+	ctrl := gomock.NewController(t)
+	store := mock.NewMockStore(ctrl)
+	var entIt *mock.MockEntriesIterator
+	entItErr := errors.New("failed to scan")
+	store.EXPECT().Scan(ctx, gomock.Any(), gomock.Any()).Return(entIt, entItErr).Times(1)
+
+	it := kv.NewPartitionIterator(ctx, store, (&graveler.StagedEntryData{}).ProtoReflect().Type(), "partitionKey")
+	it.SeekGE([]byte("test"))
+	it.Close() // verify we don't crash after see failedcx
```

Comment:
see failedcx?

Reply:
Ahh I think I understand now, it's German for "the failed seek" 😆

Reply:
fixed the comment

Reply:
my the hands write something, while the head is thinking about the next line.